spec/outputs/webhdfs_spec.rb

```diff
@@ -0,0 +1,131 @@
+# encoding: utf-8
+require 'logstash/devutils/rspec/spec_helper'
+require 'logstash/outputs/webhdfs'
+require 'webhdfs'
+require 'json'
+
+describe 'outputs/webhdfs' do
+
+  webhdfs_host = 'localhost'
+  webhdfs_port = 50070
+  webhdfs_user = 'hadoop'
+  path_to_testlog = '/user/hadoop/test.log'
+  current_logfile_name = '/user/hadoop/test.log'
+  current_config = ""
+
+      event = LogStash::Event.new(
+        'message' => 'Hello world!',
+        'source' => 'out of the blue',
+        'type' => 'generator',
+        'host' => 'localhost',
+        '@timestamp' => LogStash::Timestamp.now)
+
+  default_config =  { 'host' => webhdfs_host,
+                      'user' => webhdfs_user,
+                      'path' => path_to_testlog,
+                      'compression' => 'none' }
+
+  client = WebHDFS::Client.new(webhdfs_host, webhdfs_port, webhdfs_user)
+
+  context 'when initializing' do
+
+    it 'should fail to register without required values' do
+      expect { LogStash::Plugin.lookup("output", "webhdfs").new() }.to raise_error(error=LogStash::ConfigurationError)
+    end
+
+    it 'should register with default values' do
+      subject = LogStash::Plugin.lookup("output", "webhdfs").new(default_config)
+      expect { subject.register }.to_not raise_error
+    end
+
+    it 'should have default config values' do
+      subject = LogStash::Plugin.lookup("output", "webhdfs").new(default_config)
+      insist { subject.port } == 50070
```

Comment:
Done in https://github.com/dstore-dbap/logstash-output-webhdfs-1/commit/02696234946e21c8d66b5dd364ce941371cf6b76